src/uint/karatsuba_mul.rs

```diff
@@ -0,0 +1,414 @@
+//! Karatsuba multiplication
```

Comment:
Small nit but I'd probably suggest putting this under `mul/karatsuba.rs`

Reply:
src/uint/mul or src/mul?

Reply:
`src/uint/mul`